### ISSUE TITLE
Remove use of ONE_WEEK interval?

### ISSUE THREAD

**Issue by nivemaham** [RADAR-CNS/RADAR-Dashboard#322]
How essential is the ONE_WEEK resolution data for the dashboard? As we are experiencing extensive resource consumption from `kafka-streams`, it would be better if we could remove streams that are not very useful. Do we expect the aggregated out of ONE_WEEK would be interesting? If not I would recommend that we remove support for ONE_WEEK from dashboard and turn off those streams on back-end. 
@blootsvoets @afolarin @yatharthranjan @mpgxvii

**Comment by yatharthranjan** [comment]
Hi nivethika, on the contrary I think the 10 sec streams are more resource intensive and also the researchers don't really need the 10 sec interval. 1 min is more than enough as they just need an overview of the data.  The 10 seconds has the least commit interval and causes higher CPU utilisation. Also the 10 sec and 1 min are the only streams that rebalance sometimes and migrate to another thread. After the recent updates to optimise memory utilisation and commit interval, I think the CPU utilisation of 1 week streams has gone down a lot. 
What do you guys think?

**Comment by blootsvoets** [comment]
OK, that is good to know. If that is the case, we could look at removing the 10 second interval instead. That would also reduce pressure on the MongoDB database. If the CPU usage is still exorbitant, we could look at decreasing the serialization/deserialization cost of the stream state.

**Comment by mpgxvii** [comment]
@nivemaham Do you mean the ONE_WEEK for all data (both sensor data and aggregate/volume data) or just aggregate data? But in terms of aggregate data, the page loads ONE_DAY on the volume graph by default so I guess ONE_WEEK won’t really be used unless the volume time frame chosen would have a length of more than 3 years (in which case I also am not sure whether this huge time frame will be very informative). 

On the other hand, the advantage of the ONE_WEEK resolution is that it loads quickest since the dataset is the least for a certain time frame.

**Comment by yatharthranjan** [comment]
The CPU usage stays around 500% but that is fine as the streams are much more stable now. 
But if possible we should definitely try to bring this down.

**Comment by nivemaham** [comment]
@mpgxvii Yes. If not necessary for both sensor data and volume data. If necessary we can allow ONE_WEEK just for volume data (since counting cost a lot `less` that calculating other statistics). I do agree with you about the use of such huge time-frame. Further, given that we can change the time window of the data which we want to see, researcher can rather scroll back old data in parts. e.g. ( 2018-2017, 2017-2016, 2016-2015) rather than (2018-2015). 

@yatharthranjan That is good to know. I understand that 10 SEC cost of lost of CPU whereas I would assume the ONE_WEEK would consume more memory. We should investigate both windows and their consumption and weigh it with the actual benefit we could get.

**Comment by afolarin** [comment]
Agree, I'm not sure the current infrastructure will scale so well under present load. Would be good to do an appraisal of the different aggregation streams. But I agree the ONE_WEEK could perhaps go?
